src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42IslRttSpec.groovy

```diff
@@ -1,358 +1,649 @@
+            s42IslRttTurningRule = s42IslRttRules.find { it.cookie == Cookie.SERVER_42_ISL_RTT_TURNING_COOKIE }
         }
 
-        cleanup: "Revert system to original state"
-        revertToOrigin(islRttFeatureStartState, initialSwitchRtt)
+        when: "Update server42Port on the switch"
+        def newS42Port = originSwProps.server42Port + 1
+        northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId).jacksonCopy().tap({
+            it.server42Port = newS42Port
+        }))
+
+        then: "SERVER_42_ISL_RTT_OUTPUT_COOKIE and SERVER_42_ISL_RTT_INPUT rules updated according to the changes"
+        and: "SERVER_42_ISL_RTT_TURNING_COOKIE is not changed"
+        wait(RULES_INSTALLATION_TIME) {
+            def rules = northbound.getSwitchRules(sw.dpId).flowEntries.findAll {
+                (it.cookie in [Cookie.SERVER_42_ISL_RTT_TURNING_COOKIE, Cookie.SERVER_42_ISL_RTT_OUTPUT_COOKIE]) ||
+                        (new Cookie(it.cookie).getType() in [CookieType.SERVER_42_ISL_RTT_INPUT])
+            }
+            assert rules.size() == northbound.getLinks(sw.dpId, null, null, null).size() + 2
+            assert rules.findAll {
+                new Cookie(it.cookie).getType() == CookieType.SERVER_42_ISL_RTT_INPUT
+            }*.match.inPort.unique() == [newS42Port.toString()]
+            assert rules.find {
+                it.cookie == Cookie.SERVER_42_ISL_RTT_OUTPUT_COOKIE
+            }.instructions.applyActions.flowOutput == newS42Port.toString()
+            assert expect(rules.find { it.cookie == Cookie.SERVER_42_ISL_RTT_TURNING_COOKIE }, sameBeanAs(s42IslRttTurningRule)
+                    .ignoring("byteCount")
+                    .ignoring("packetCount")
+                    .ignoring("durationSeconds")
+                    .ignoring("durationNanoSeconds"))
+        }
+
+        when: "server42IslRtt feature toggle is set to false"
+        changeIslRttToggle(false)
+
+        and: "server42IslRtt is set to AUTO on the switch"
+        northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId).jacksonCopy()
+                .tap({ it.server42IslRtt = RttState.AUTO.toString() }))
+
+        then: "No IslRtt rules on the switch"
+        wait(RULES_DELETION_TIME) { checkIslRttRules(sw.dpId, false) }
+
+        when: "server42IslRtt feature toggle is set true"
+        changeIslRttToggle(true)
+
+        then: "IslRtt rules are installed"
+        wait(RULES_INSTALLATION_TIME) { checkIslRttRules(sw.dpId, true) }
+
+        when: "Update server42Port on the switch(revert to origin)"
+        northbound.updateSwitchProperties(sw.dpId, northbound.getSwitchProperties(sw.dpId).jacksonCopy()
+                .tap({ it.server42Port = originSwProps.server42Port }))
+
+        then: "SERVER_42_ISL_RTT_OUTPUT_COOKIE and SERVER_42_ISL_RTT_INPUT rules updated according to the changes"
```

Comment:
How do you know which exactly state is 'origin' to do the check?

Reply:
I retrieve it before.
I mean that I found the switch which is connected to s42 and got swProps from it `def originSwProps = northbound.getSwitchProperties(sw.dpId)`. Assume that this data is correct(origin).